File: 面试题06-从尾到头打印链表.py
class Solution:
    def reversePrint(self, head: ListNode) -> List[int]:
        mylist = []
        while head:
            mylist.append(head.val)
            head = head.next

        i = 0
        j = len(mylist) - 1
        while i < j:
            mylist[i], mylist[j] = mylist[j], mylist[i]
            i += 1
            j -= 1
        
        return mylist

File: 面试题06-从尾到头打印链表.cpp
class Solution {
public:
    vector<int> reversePrint(ListNode* head) {
        vector<int> result;
        while(head){
            result.push_back(head->val);
            head = head->next;
        }

        int i = -1, j = result.size();
        while(++i < --j){
            swap(result[i], result[j]);
        }

        return result;
    }
};
